plugin/iamutil/iam_resource.go

```diff
@@ -74,7 +74,15 @@ func (r *parsedIamResource) SetIamPolicyRequest(p *Policy) (req *http.Request, e
 	return r.constructRequest(&r.config.SetMethod, strings.NewReader(reqJson))
 }
 
+var requestPolicyVersion3 = `{"options": {"requestedPolicyVersion": 3}}`
+
 func (r *parsedIamResource) GetIamPolicyRequest() (*http.Request, error) {
+	// In order to support Resource Manager policies with conditional bindings,
+	// we need to request the policy version of 3. This request parameter is backwards compatible
+	// and will return version 1 policies if they are not yet updated to version 3.
+	if r.config != nil && r.config.Service == "cloudresourcemanager" {
```

Comment:
I understand, it's definitely a one-off check as there are [other services](https://cloud.google.com/iam/docs/conditions-overview#resources) that aren't included that also support IAM conditions.  

I wasn't sure how to scrape that information from the docs into the [`generatedIamResources`](https://github.com/hashicorp/vault-plugin-secrets-gcp/blob/master/plugin/iamutil/iam_resources_generated.go#L8), perhaps it could be hardcoded in the generation tool as a service map like [`correctedRequestFormats`](https://github.com/hashicorp/vault-plugin-secrets-gcp/blob/master/plugin/iamutil/internal/generate_iam.go#L39). And then `LatestPolicyVersion` could be added as an optional field in [`IamRestResource`](https://github.com/hashicorp/vault-plugin-secrets-gcp/blob/master/plugin/iamutil/iam_resource.go#L26) which could be used to replace this specific service conditional.